client/my-sites/themes/controller.jsx

```diff
@@ -77,19 +91,51 @@ export function loggedOut( context, next ) {
 	next();
 }
 
-export function fetchThemeData( context, next ) {
+export function fetchThemeData( context, next, shouldUseCache = false ) {
 	if ( ! context.isServerSide ) {
 		return next();
 	}
 
 	const queryParams = {
 		search: context.query.s,
 		tier: context.params.tier,
-		filter: context.params.filter,
+		filter: compact( [ context.params.filter, context.params.vertical ] ).join( ',' ),
 		page: 0,
 		perPage: PER_PAGE,
 	};
+	const cacheKey = context.path;
 
 	context.store.dispatch( query( queryParams ) );
-	context.store.dispatch( fetchNextPage( false ) ).then( () => next() );
+	context.store.dispatch( incrementThemesPage( false ) );
+
+	if ( shouldUseCache ) {
+		const cachedData = themesQueryCache.get( cacheKey );
+		if ( cachedData ) {
+			debug( `found theme data in cache key=${ cacheKey }` );
+			context.store.dispatch( cachedData.action );
+			context.renderCacheKey = context.path + cachedData.timestamp;
+			return next();
+		}
+	}
+
+	context.store.dispatch( fetchThemes( false ) )
```

Comment:
> I'm happy with a default param solution to make this less confusing — but just noting that there's a lot of other action creators that rely on site, so should this be a universal solution everywhere?

So this is an interesting question. I think we're heading towards checking for `null/undefined` for sites, since our Redux selectors now return `null` instead of `false` for unknown sites, converse to what we have with `sites-list#getSelectedSite`.

I would be happy to defer this to a follow up PR, if we're going to, or want to make more adjustments. What do you think?